src/main/java/ch/njol/skript/expressions/ExprDifference.java

```diff
@@ -33,123 +28,168 @@
-		"\tmessage \"You have to wait a minute before using this command again!\""})
+@Examples({
+	"if difference between {command::%player%::lastuse} and now is smaller than a minute:",
+	"\tmessage \"You have to wait a minute before using this command again!\""
+})
 @Since("1.4")
 public class ExprDifference extends SimpleExpression<Object> {
 	
 	static {
-		Skript.registerExpression(ExprDifference.class, Object.class, ExpressionType.COMBINED, "difference (between|of) %object% and %object%");
+		Skript.registerExpression(ExprDifference.class, Object.class, ExpressionType.COMBINED,
+			"difference (between|of) %object% and %object%"
+		);
 	}
 	
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<?> first, second;
-	
-	@SuppressWarnings("rawtypes")
+
 	@Nullable
+	@SuppressWarnings("rawtypes")
 	private Arithmetic math;
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Class<?> relativeType;
-	
-	@SuppressWarnings({"unchecked", "null", "unused"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		first = exprs[0];
-		second = exprs[1];
-		final ClassInfo<?> ci;
-		if (first instanceof Variable && second instanceof Variable) {
-			ci = DefaultClasses.OBJECT;
-		} else if (first instanceof Literal<?> && second instanceof Literal<?>) {
-			first = first.getConvertedExpression(Object.class);
-			second = second.getConvertedExpression(Object.class);
-			if (first == null || second == null)
-				return false;
-			ci = Classes.getSuperClassInfo(Utils.getSuperType(first.getReturnType(), second.getReturnType()));
-		} else {
-			if (first instanceof Literal<?>) {
-				first = first.getConvertedExpression(second.getReturnType());
-				if (first == null)
-					return false;
-			} else if (second instanceof Literal<?>) {
-				second = second.getConvertedExpression(first.getReturnType());
-				if (second == null)
-					return false;
-			}
-			if (first instanceof Variable) {
-				first = first.getConvertedExpression(second.getReturnType());
-			} else if (second instanceof Variable) {
-				second = second.getConvertedExpression(first.getReturnType());
-			}
-			assert first != null && second != null;
-			ci = Classes.getSuperClassInfo(Utils.getSuperType(first.getReturnType(), second.getReturnType()));
-		}
-		assert ci != null;
-		if (!ci.getC().equals(Object.class) && ci.getMath() == null) {
-			Skript.error("Can't get the difference of " + CondCompare.f(first) + " and " + CondCompare.f(second), ErrorQuality.SEMANTIC_ERROR);
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		Expression<?> first = LiteralUtils.defendExpression(exprs[0]);
+		Expression<?> second = LiteralUtils.defendExpression(exprs[1]);
+		if (!LiteralUtils.canInitSafely(first, second)) {
```

Comment:
Doesn't canInitSafely go before defending?

Reply:
No, IIRC canInitSafely is to make sure that defending them was actually successful